src/mcmc/Inference.jl

```diff
@@ -115,6 +115,23 @@ DynamicPPL.unflatten(vi::SimpleVarInfo, θ::NamedTuple) = SimpleVarInfo(θ, vi.l
 # Algorithm for sampling from the prior
 struct Prior <: InferenceAlgorithm end
 
+function AbstractMCMC.step(
+    rng::Random.AbstractRNG,
+    model::DynamicPPL.Model,
+    sampler::DynamicPPL.Sampler{<:Prior},
+    state=nothing;
+    kwargs...,
+)
+    vi = last(DynamicPPL.evaluate!!(
+        model,
+        VarInfo(),
+        SamplingContext(
+            rng, DynamicPPL.SampleFromPrior(), DynamicPPL.PriorContext()
```

Comment:
Ideally yes, but I don't think that's going to work atm because `SampleFromPrior` is used in too many different contexts as just "do a first sample" or "sample but also compute logjoint" or "don't sample, just evaluate the model for these parameters because I've set the `delete` flag to `false`"...

Reply:
Indeed! This is related to https://github.com/TuringLang/DynamicPPL.jl/issues/573